### ISSUE TITLE
Add dts support to SPI slave drivers & tests

### ISSUE THREAD

**Issue by galak** [zephyrproject-rtos/zephyr#10894]
Issue #10636 exposed a problem that a few cases that SPI slave drivers aren't converted to DTS.  Doing a grep in the tree for SPI_._NAME or SPI_DRV_NAME or spi_config we come up with:

```
tests/drivers/spi/spi_loopback/Kconfig:	default SPI_1_NAME


```

We need to convert these slave drivers to use DTS.

> I'm working on it. So far I found only one thing that is missing on the master side (in `drivers/bluetooth/hci/spi.c`). The CS line [is not asserted](https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/bluetooth/hci/spi.c#L131) in the general case, there is only some specific steering of it for BlueNRG. But it's not that trivial to get this information from DTS since this line may be (at least for Nordic SoCs it is) configured in `cs-gpios` property in the parent node. Configuration of other settings is already in place (through [the "zephyr,bt-hci-spi" node](https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/arm/96b_carbon/96b_carbon.dts#L89)).
> On the slave side (in `samples/bluetooth/hci_spi`) to get [the relevant settings](https://github.com/zephyrproject-rtos/zephyr/blob/master/samples/bluetooth/hci_spi/src/main.c#L51) I will follow the way you started in [galak@f30bc9e](https://github.com/galak/zephyr/commit/f30bc9ea7033a1b63d5dde87ae765f2130050128), but I think that instead of adding the `spi-slave` property and setting `#address-cells` to `0` I will use a separate binding for "nordic,nrf-spis" node that will inherit from "spi-device". But first I need to make the hci_spi example working. Currently it is not handling correctly SPI transactions as it does not configure SPI to work in the slave mode, so the whole communication does not start. But even after I corrected this I still get some "invalid message type" errors. I need to debug it further.

Any update on this?

**Comment by galak** [comment]
@vanwinkeljan can you look at adding DTS support to drivers/display/display_ili9340.c.  Thanks.

**Comment by galak** [comment]
@mbolivar can you add dts support to WS2812 & LPD880x

**Comment by galak** [comment]
@tbursztyka can you look at converting adc_ti_adc108s102 to device tree.

**Comment by galak** [comment]
> @mbolivar can you add dts support to WS2812 & LPD880x

Any update?  This drivers will soon not be usable once all SPI controllers use DTS.

**Comment by galak** [comment]
@agansari can you look at converting the eth_enc28j60 driver over to use DTS.

**Comment by galak** [comment]
PR #11165 to convert over BMI160 Sensor

**Comment by carlescufi** [comment]
@anangl  have you taken a look at `drivers/bluetooth/hci/spi.c` ?

**Comment by galak** [comment]
> @anangl have you taken a look at `drivers/bluetooth/hci/spi.c` ?

@anangl / @carlescufi any update?

**Comment by carlescufi** [comment]
@galak not from me, no. I hope that @anangl can update us.

**Comment by anangl** [comment]
@galak
 
> > @anangl have you taken a look at `drivers/bluetooth/hci/spi.c` ?
> 
> @anangl / @carlescufi any update?

I'm working on it. So far I found only one thing that is missing on the master side (in `drivers/bluetooth/hci/spi.c`). The CS line [is not asserted](https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/bluetooth/hci/spi.c#L131) in the general case, there is only some specific steering of it for BlueNRG. But it's not that trivial to get this information from DTS since this line may be (at least for Nordic SoCs it is) configured in `cs-gpios` property in the parent node. Configuration of other settings is already in place (through [the "zephyr,bt-hci-spi" node](https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/arm/96b_carbon/96b_carbon.dts#L89)).
On the slave side (in `samples/bluetooth/hci_spi`) to get [the relevant settings](https://github.com/zephyrproject-rtos/zephyr/blob/master/samples/bluetooth/hci_spi/src/main.c#L51) I will follow the way you started in https://github.com/galak/zephyr/commit/f30bc9ea7033a1b63d5dde87ae765f2130050128, but I think that instead of adding the `spi-slave` property and setting `#address-cells` to `0` I will use a separate binding for "nordic,nrf-spis" node that will inherit from "spi-device". But first I need to make the hci_spi example working. Currently it is not handling correctly SPI transactions as it does not configure SPI to work in the slave mode, so the whole communication does not start. But even after I corrected this I still get some "invalid message type" errors. I need to debug it further.
